test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -1707,31 +1703,18 @@ BOOST_AUTO_TEST_CASE(transfer_ether)
 				throw;
 			}
 		}
-
-		contract D {
-			// This takes about ~3600 gas, which exceeds the 2300 gas stipend.
-			function () payable {
-				bytes32 tmp = 1;
-				for (uint i = 0; i < 20; i++)
-					tmp = sha3(tmp);
```

Comment:
I don't think it is good pattern to spend a sender's money to ultimately reject a token transfer to you. Also I think the concurrency model works much better if you cannot reject a transfer. We should take a look at how this is done in regular bookkeeping. I would say it is better to first take the transfer into a "pending" contract which can then ultimately either send it back or send it to the "accepted" contract.